pyvrp/search/neighbourhood.py

```diff
@@ -133,17 +133,17 @@ def _compute_proximity(
     """
     clients = [data.client(idx) for idx in range(data.num_clients + 1)]
 
-    early = np.array([client.tw_early for client in clients])
-    late = np.array([client.tw_late for client in clients])
-    service = np.array([client.service_duration for client in clients])
-    prize = np.array([client.prize for client in clients])
+    early = np.asarray([client.tw_early for client in clients])
+    late = np.asarray([client.tw_late for client in clients])
+    service = np.asarray([client.service_duration for client in clients])
+    prize = np.asarray([client.prize for client in clients])
     duration = np.asarray(data.duration_matrix(), dtype=float)
 
     min_wait_time = early[..., :] - duration - service[:, ...] - late[:, ...]
     min_time_warp = early[:, ...] + service[:, ...] + duration - late[..., :]
 
     return (
-        data.distance_matrix()
+        np.asarray(data.distance_matrix(), dtype=float)
```

Comment:
@leonlan this appears to have been it!

Reply:
Nice job! Though I don't really understand what went wrong here and how it's fixed 😅

Reply:
I don't really get this one either tbh. But the previously problematic instance completes after this and I cannot be bothered right now to get to the bottom it.

(it might still not work if it's just postponed the problem to appear in an even later iteration, but the benchmark I'm currently running will let us know if that's the case)